### PR TITLE
Issue/43/cleanup

### PR DESCRIPTION
**Pull request by eacharles**
- Remove setting of support ranges from various distributions, this was forcing scipy to unroll the arrays and complicated the code.  We just set the PDFs to zero outside the range where the interpolation or histogram is defined.
- Cleaned up doc, moving old docs to 'old_docs' directory
- Cleaned up class test_data, so that it is defined in functions rather than statically, to avoid computations on module import.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,8 +21,7 @@ To make sure you are running the latest version, you can rebuild the Binder `qp`
 
 You can also browse the (un-executed) notebooks here in this repo:
 
-* **[Basic  Demo](http://htmlpreview.github.io/?https://github.com/aimalz/qp/blob/html/demo.html)** [(raw notebook)](https://github.com/aimalz/qp/blob/master/docs/notebooks/demo.ipynb)
-* **[KL Divergence  Illustration](http://htmlpreview.github.io/?https://github.com/aimalz/qp/blob/html/kld.html)** [(raw notebook)](https://github.com/aimalz/qp/blob/master/docs/notebooks/kld.ipynb)
+* **[Basic  Demo](http://htmlpreview.github.io/?https://github.com/LSSTDESC/qp/blob/html/demo.html)** [(raw notebook)](https://github.com/LSSTDESC/qp/blob/master/docs/notebooks/demo.ipynb)
```
Comment (sschmidt23) [inline README.md:24]: I'm getting a 404:Not Found for this link, did your rearrangement of the locations of the notebooks switch this around somehow?

**Comment by eacharles** [comment]
More or less yeah, I have rendered the new versions of the NB to html.

Do you happen to know how to do that?

-e

File: README.md (hunk 1 above)
Comment (sschmidt23) [reply README.md:24]: Oh, whoops, I saw demo.html in "old_docs" and thought it was the old "pre-fork" demo without all of the updates. I see now that it is actually the updated file, so 
http://htmlpreview.github.io/?https://github.com/LSSTDESC/qp/blob/issue/43/cleanup/old_docs/notebooks/demo.html
is a working link (we'll obviously want to switch the branch we're pointing to to  master for the commit, so
http://htmlpreview.github.io/?https://github.com/LSSTDESC/qp/blob/master/old_docs/notebooks/demo.html
would be the proper link).  However, is `old_docs` the best location for the rendered notebook, or should we maybe move it to the `nb` colocated with the actual ipynb file?

Comment (sschmidt23) [reply README.md:24]: Hmm, or if the copy in `old_docs` is not the newest, did you forget to include the new rendered html version of the notebook when you pushed?  I don't see any other copies of demo.html

**Comment by eacharles** [comment]
I just blindly moved everything to old docs then tried to grab just the stuff I wanted.   Namely the rst files & two example notebooks.

Now that I think about it, this was sure to break something.

-e

File: README.md (hunk 1 above)
Comment (sschmidt23) [reply README.md]: new rendered notebooks copied to docs, we'll deal with rendering in an action in a future PR

**Review by sschmidt23 — APPROVED** [review]
Changes look good, ready to merge.

**Pull request merged**